packages/core/src/internal/temporal-util.ts

```diff
@@ -288,16 +288,7 @@ export function dateToIsoString (
   month: NumberOrInteger | string,
   day: NumberOrInteger | string
 ): string {
-  year = int(year)
-  const isNegative = year.isNegative()
-  if (isNegative) {
-    year = year.multiply(-1)
-  }
-  let yearString = formatNumber(year, 4)
-  if (isNegative) {
-    yearString = '-' + yearString
-  }
-
+  const yearString = formatNumber(year, 6, { usePositiveSign: true })
```

Comment:
Is there a reason not to use `formatNumber(year, 4)` for years between 0 and 9999? This would increase the compatibility with other ISO implementations.

Reply:
The issue:

```
> new Date('0099-07-04')
Sat Jul 04 0099 00:50:20 GMT+0050 (Central European Summer Time)
> new Date('+0099-07-04')
Sun Jul 04 1999 00:00:00 GMT+0200 (Central European Summer Time)
> new Date('-0099-07-04')
Sun Jul 04 1999 00:00:00 GMT+0200 (Central European Summer Time)

Reply:
Oh. I didn't know JS also did this with strings. Some JS design decisions just make we want to shake my head.